swift/extractor/infra/TargetDomains.cpp

```diff
@@ -42,4 +46,33 @@ std::optional<TrapDomain> createTargetTrapDomain(SwiftExtractorState& state,
   }
   return std::nullopt;
 }
+
+std::optional<LinkDomain> createTargetLinkDomain(const SwiftExtractorState& state,
+                                                 const std::filesystem::path& target) {
+  if (target.empty()) {
+    return std::nullopt;
+  }
+  auto file = getRelativeTrapPath(target, TrapType::linkage, ".link");
+  auto ret =
+      TargetFile::create(file, state.configuration.trapDir, state.configuration.getTempTrapDir());
+  if (ret) {
+    return LinkDomain{*std::move(ret)};
+  }
+  return std::nullopt;
+}
+
+std::optional<ObjectDomain> createTargetObjectDomain(const SwiftExtractorState& state,
+                                                     const std::filesystem::path& target) {
+  if (target.empty()) {
+    return std::nullopt;
+  }
+  auto file = getRelativeTrapPath(target, TrapType::linkage, ".odep");
+  auto ret =
+      TargetFile::create(file, state.configuration.trapDir, state.configuration.getTempTrapDir());
+  if (ret) {
+    return ObjectDomain{*std::move(ret)};
+  }
+  return std::nullopt;
+}
```

Comment:
Nit: it seems these two methods can be squashed into one with some (template) arguments?

Reply:
I'm surprised at being suggested to put _more_ templates in my code :rofl: